datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxCustomController.java

```diff
@@ -0,0 +1,45 @@
+package com.sequenceiq.datalake.controller.sdx;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.authorization.annotation.CheckPermissionByAccount;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.metric.MetricType;
+import com.sequenceiq.datalake.metric.SdxMetricService;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+import com.sequenceiq.sdx.api.endpoint.SdxCustomEndpoint;
+import com.sequenceiq.sdx.api.model.SdxClusterResponse;
+import com.sequenceiq.sdx.api.model.SdxCustomClusterRequest;
+
+@Controller
+public class SdxCustomController implements SdxCustomEndpoint {
+
+    @Inject
+    private SdxService sdxService;
+
+    @Inject
+    private SdxClusterConverter sdxClusterConverter;
+
+    @Inject
+    private SdxMetricService metricService;
+
+    @Override
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.CREATE_DATALAKE)
+    public SdxClusterResponse create(String name, @Valid SdxCustomClusterRequest createSdxClusterRequest) {
+        String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
+        Pair<SdxCluster, FlowIdentifier> result = sdxService.createSdx(userCrn, name, createSdxClusterRequest, null);
+        SdxCluster sdxCluster = result.getLeft();
+        metricService.incrementMetricCounter(MetricType.INTERNAL_SDX_REQUESTED, sdxCluster);
```

Comment:
Do not we possibly need some either kind of metric here?

Reply:
I added a new metric type CUSTOM_SDX_REQUESTED to record the metric